deploy/teardown_resources.sh

```diff
@@ -7,31 +7,59 @@ set -euxo pipefail
 set_namespace $CONJUR_NAMESPACE_NAME
 
 configure_cli_pod
+
+# Helm Chart clean-up
+rm -f conjur.pem
+
 if [ "${DEV}" = "false" ]; then
-  $cli_with_timeout "exec $(get_conjur_cli_pod_name) -- conjur variable values add secrets/test_secret \"supersecret\""
+  pushd ../../../secrets-provider/ci
+else
+  pushd ../secrets-provider/ci
+fi
+    find . -type f ! -name '*template.yaml' -delete
+popd
+
+# Delete Helm Chart if already exists
+set_namespace $APP_NAMESPACE_NAME
+if [ "$(helm ls -aq | wc -l | tr -d ' ')" != 0 ]; then
+  helm delete $(helm ls -aq)
 fi
 
+set_namespace $CONJUR_NAMESPACE_NAME
+
+$cli_with_timeout "exec $(get_conjur_cli_pod_name) -- conjur variable values add secrets/test_secret \"supersecret\""
+
 set_namespace $APP_NAMESPACE_NAME
 
 $cli_with_timeout "delete secret dockerpullsecret --ignore-not-found=true"
 
 $cli_with_timeout "delete clusterrole secrets-access-${UNIQUE_TEST_ID} --ignore-not-found=true"
 
+$cli_with_timeout "delete role another-secrets-provider-role --ignore-not-found=true"
```

Comment:
i may have asked this elsewhere but i can't find it - why do we need 2 of each resource? what is the purpose of this?

Reply:
In short, we have some tests that test two secrets providers (`TEST_ID_18_helm_multiple_provider_multiple_secrets.sh`)  and two secrets providers same secret (`TEST_ID_19_helm_multiple_provider_same_secret.sh`) and two k8s secrets (`TEST_ID_20_helm_service_account_does_not_exist.sh`)
2 k8s resources cannot have the same role,rolebinding,serviceaccount,etc so we need to add two of everything for those ones

Reply:
why can't they have the same role and service account?

Reply:
asked this question to @eladkug and we agreed on two of everything

Reply:
i'd like to challenge that decision. why can't the secrets-providers have the same service account and role? isn't it closer to a customer's environment?

i also want to challenge the decision to have only one file for a secret that is used for 2 secrets. i think that having 2 files is more explicit to what you are trying to do.

Reply:
@eladkug please address Oren's first point

@orenbm I am not understanding why having a single files is an issue here. I prefer it being cleaner and having a single one that is dynamically updated and so that we don't have 2 places to clean up and two functions to create the secret etc. Earlier you mentioned why we have two of everything and you were right. I think where we can make resources reused and dynamically created we should because it will lead to a cleaner

Reply:
As we raised it before its different scenarios.

Reply:
the multiple files issue is not a blocker.

Reply:
agree